packages/client/src/api/index.js

```diff
@@ -1,11 +1,12 @@
-import { authenticate } from "./authenticate"
-// import appStore from "../state/store"
+import {authenticate} from "./authenticate"
+import {getAppIdFromPath} from "../render/getAppId";
 
 const apiCall = method => async ({ url, body }) => {
   const response = await fetch(url, {
     method: method,
     headers: {
       "Content-Type": "application/json",
+      "x-budibase-app-id": getAppIdFromPath(),
```

Comment:
Can we always rely on this? Or will the middleware on the server catch it if it's not set?

Reply:
Honestly I'm not sure if we can rely on this fully, I've added a check to the `getAppIdFromPath` as such:
```
export const getAppIdFromPath = () => {
  let appId = location.pathname.split("/")[1]
  return appId.startsWith("app_") ? appId : undefined
}
```
But I'm not sure what the best way to handle that the "appId" is infact correct is, whether to always send up that part of the path and then have the server check whether or not its valid, if its not continue its search beyond the header. I feel like not setting the header is likely the best way to approach this if the appId is invalid, otherwise things are going to get mighty confusing.